sapi/apache2handler/php_functions.c

```diff
@@ -363,21 +363,18 @@ PHP_MINFO_FUNCTION(apache)
 #endif
 
 	for (n = 0; ap_loaded_modules[n]; ++n) {
-		char *s = (char *) ap_loaded_modules[n]->name;
+		const char *s = ap_loaded_modules[n]->name;
 		if ((p = strchr(s, '.'))) {
 			smart_str_appendl(&tmp1, s, (p - s));
 		} else {
 			smart_str_appends(&tmp1, s);
 		}
 		smart_str_appendc(&tmp1, ' ');
 	}
-	if (tmp1.s) {
```

Comment:
Note that we now have a trailing ' '. Although this code looks dubious to begin with (len > 0 check but still setting index 0 :thinking: ).

Reply:
it was unconditionally regardless below for displaying

Reply:
Lol, indeed, even more bizarre... That said, it's maybe best to keep the behaviour that strips the last ' '.